lightning/src/ln/peer_handler.rs

```diff
@@ -1084,22 +1084,22 @@ impl<Descriptor: SocketDescriptor, CM: Deref> PeerManager<Descriptor, CM> where
 						},
 						None => {}
 					}
+				} else if !peer.channel_encryptor.is_ready_for_encryption() {
+					// The peer needs to complete its handshake before we can exchange messages
+					return true;
 				}
 
 				let ping = msgs::Ping {
 					ponglen: 0,
 					byteslen: 64,
 				};
-				peer.pending_outbound_buffer.push_back(encode_msg!(&ping));
+				peer.pending_outbound_buffer.push_back(peer.channel_encryptor.encrypt_message(&encode_msg!(&ping)));
+
 				let mut descriptor_clone = descriptor.clone();
 				self.do_attempt_write_data(&mut descriptor_clone, peer);
 
-				if peer.awaiting_pong {
-					false // Drop the peer
-				} else {
-					peer.awaiting_pong = true;
-					true
-				}
+				peer.awaiting_pong = !peer.awaiting_pong; // flip the condition
```

Comment:
This is pretty hard to read, can you just drop the peer at the top of this fn instead of waiting to the end and returning the awaiting_pong var?